cluster-operator/src/main/java/io/strimzi/operator/cluster/operator/assembly/KafkaAssemblyOperator.java

```diff
@@ -177,6 +190,8 @@ public KafkaAssemblyOperator(Vertx vertx, boolean isOpenShift,
         private ConfigMap kafkaMetricsAndLogsConfigMap;
         private ReconcileResult<StatefulSet> kafkaDiffs;
         private boolean kafkaForcedRestart;
+        private String kafkaExternalBootstrapAddress;
+        private Map<String, String> kafkaExternalAddresses = new TreeMap<>();
```

Comment:
Why does this need to be ordered?

Reply:
OK, I see now. Neat. In which case `generateStatefulSet()` should take a `SortedMap<Integer, String>` rather than a `Map<String, String>`. `SortedMap` because that's what is required for it to work, and `String` because I suspect it might break right now with > 9 brokers.

Reply:
Good catch. You are right. Thanks for seeing this.